File: 10/01/PageObjects/DynamicLoadingPage.cs
using OpenQA.Selenium;

namespace PageObjects 
{
    class DynamicLoadingPage : BasePage
    {
        By StartButton = By.CssSelector("#start > button");
        By FinishText = By.Id("finish");

        public DynamicLoadingPage(IWebDriver driver) : base(driver) { }

        public void LoadExample(int exampleNumber)
        {
            Visit("http://the-internet.herokuapp.com/dynamic_loading/" + exampleNumber);
            Click(StartButton);
        }

        public bool FinishTextPresent()
        {
            return IsDisplayed(FinishText, 10);
        }
    }
}

File: code-examples/javascript/11/02/pages/DynamicLoadingPage.js
const BasePage = require('./BasePage')

const START_BUTTON = { css: '#start button' }
const FINISH_TEXT = { id: 'finish' }

class DynamicLoadingPage extends BasePage {
  constructor(driver) {
    super(driver)
  }

  async loadExample(exampleNumber) {
    await this.visit('/dynamic_loading/' + exampleNumber)
    await this.click(START_BUTTON)
  }

  async isFinishTextPresent() {
    return this.isDisplayed(FINISH_TEXT, 10000)
  }
}

module.exports = DynamicLoadingPage
